Bookmark

zhenlineo opened the pull request:
Based on https://github.com/neo4j/neo4j-dotnet-driver/pull/119

Commit:
Add server info on result summary

Commit:
Adding bookmark
Missing test to verify bookmark is set correctly

Commit:
Fix a bug where a user failed to get a Summary if an error is thrown in Consume method.

Commit:
Added a IT to ensure bookmark work as expected.
Refactored the previous ConnectionIT (now DirectDriverIT)

lutovich reviewed (changes requested):
@zhenlineo changes look fine but I think we need more tests. it seems `BookmarkCollector` class and `ISession.BeginTransaction()` method with non-null bookmark are not tested.

Commit:
Fix after code review
Added a test for using bookmark with a single instance
Fixed the bug where bookmark is not sent to server at all :D
